src/main.rs

```diff
@@ -154,16 +154,34 @@ enum Sevctl {
 }
 
 fn download(
-    rsp: reqwest::Result<reqwest::blocking::Response>,
+    url: &str,
     usage: Usage,
 ) -> Result<sev::Certificate> {
-    let rsp = rsp.context(format!("unable to contact {} server", usage))?;
+    let mut http_request_count = 1;
+    let mut time_wait_seconds = 3;
+    let rsp = reqwest::blocking::get(url);
+    let mut rsp = rsp.context(format!("unable to contact {} server", usage))?;
+    let _status = loop {
+
+        let rsp_status = rsp.status();
+        if rsp_status.is_success() {
+            break rsp_status;
+        } else if rsp_status.is_client_error() {
+            http_request_count += 1;
+        }
+
 
-    let status = rsp.status();
-    let mut rsp = rsp.error_for_status().context(&format!(
-        "received failure from {} server: {}",
-        usage, status
-    ))?;
+        // There is only a max of 5 HTTP requests allowed, if success is not reached within 5
+        // attempts, terminate program with proper failure message.
+        if http_request_count < 5 {
+            std::thread::sleep(std::time::Duration::new(time_wait_seconds, 0));
+            time_wait_seconds += 3;
+            let retry = reqwest::blocking::get(url);
+            rsp = retry.context(format!("unable to contact {} server", usage))?;
+        } else {
+                panic!("received failure from {} server: {}", usage, rsp_status);
```

Comment:
Don't panic. Return an error. There's a method called `context` that you can call on `Result` types to add a more human-readable error message on top of the error that you encountered.